concurrent_map.go

```diff
@@ -302,6 +349,43 @@ func (m ConcurrentMap) MarshalJSON() ([]byte, error) {
 	return json.Marshal(tmp)
 }
 
+// isEvictionOccurred check if the map size overflow the maximum allowed size.
+func (m ConcurrentMap) isEvictionOccurred(key string) bool {
+	shard := m.GetShard(key)
+	if m.Count() > shard.maxSize {
```

Comment:
this can be optimized in the following way:
1. should check only the shard's length
2. should be mutex free as it is only used internally (and should be called in a mutex protected area)

Reply:
This is already mutex free.